src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -670,9 +673,14 @@ protected override TypeSymbol InferTypeOfVarVariable(DiagnosticBag diagnostics)
                         throw ExceptionUtilities.UnexpectedValue(_deconstruction.Kind());
                 }
 
-                TypeSymbol result = this._type;
-                Debug.Assert((object)result != null);
-                return result;
+                if ((object)this._type == null)
+                {
+                    diagnostics.Add(this.ForbiddenDiagnostic, Locations[0], Name);
```

Comment:
>diagnostics.Add(this.ForbiddenDiagnostic, Locations[0], Name); [](start = 20, length = 62)

It is not clear why ```ForbiddenDiagnostic``` is an appropriate diagnostics to use in this case. Also, could you please point me to the test(s) covering this line?